### PR TITLE
supervisor frontend

### PR DESCRIPTION
**Pull request by akosyakov**
- [x] /werft ws-feature-flags=registry_facade
- [x] /werft https

#### How to test

- Check that one can still open Theia:
https://ak-ide-host.staging.gitpod-dev.com#https://github.com/gitpod-io/go-gin-app
- Check that Code is opened if .gitpod.yml contains `ide: code`:
https://ak-ide-host.staging.gitpod-dev.com#https://github.com/akosyakov/go-gin-app

### CODE REVIEW

File: components/ide/code/leeway.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    && sudo apt-get -y install --no-install-recommends libxkbfile-dev pkg-config libsecret-1-dev libxss1 dbus xvfb libgtk-3-0 libgbm1 \
+    # Clean up
+    && sudo apt-get autoremove -y \
+    && sudo apt-get clean -y \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN git clone https://github.com/gitpod-io/vscode.git --branch gp-code --single-branch gp-code
+WORKDIR /gp-code
+RUN yarn
+RUN yarn gulp gitpod
```
Comment (akosyakov) [inline components/ide/code/leeway.Dockerfile:36]: it should be `gitpod-min` for production

**Review by JesterOrNot — COMMENTED** [review]
I'm super excited! Fantastic work!

A few things I noticed

1. the `ide` property is not available for autocomplete on either vscode or theia
2. there is no documentation as to add additional IDEs in the future
3. When launching a vscode instance everything goes white and needs to be refreshed when the workspace is created
4. nitpick: Why `code` and not `vscode`?
5. Some extensions refuse to install such as `vscode-icons` due to an error "Cannot install 'vscode-icons' because this extension has defined that it cannot run on the remote server."

**Comment by akosyakov** [comment]
@JesterOrNot that's just a first step, nothing complete and official. `vscode` is a trade mark of Microsoft, I doubt we can use to name IDE bundled with Gitpod.

**Review by csweichel — APPROVED** [review]
Theia still works - LGTM

**Pull request merged**